lib/engine/game/g_18_norway/game.rb

```diff
@@ -366,6 +367,67 @@ def check_route_token(route, token)
 
           raise NoToken, 'Route must contain token' unless token
         end
+
+        def issuable_shares(entity)
+          return [] unless entity.corporation?
+          return [] unless round.steps.find { |step| step.instance_of?(G18Norway::Step::IssueShares) }.active?
+
+          num_shares = entity.num_player_shares - entity.num_market_shares
+          bundles = bundles_for_corporation(entity, entity)
+          share_price = stock_market.find_share_price(entity, :left).price
+
+          bundles
+            .each { |bundle| bundle.share_price = share_price }
+            .reject { |bundle| bundle.num_shares > num_shares }
+        end
+
+        def redeemable_shares(entity)
+          return [] unless entity.corporation?
+          return [] unless round.steps.find { |step| step.instance_of?(G18Norway::Step::IssueShares) }.active?
+
+          share_price = stock_market.find_share_price(entity, :right).price
+
+          bundles_for_corporation(share_pool, entity)
+            .each { |bundle| bundle.share_price = share_price }
+            .reject { |bundle| entity.cash < bundle.price }
+        end
+
+        def emergency_issuable_bundles(corp)
+          return [] if corp.trains.any?
```

Comment:
We use `!empty?` instead of `any?` as it compiles to more efficient JavaScript. `any?` gives a callback function that is called iteratively for the collection, whereas `empty?` can just look at the size of the array.

```suggestion
          return [] unless corp.trains.empty?
```